src/lib/CollisionPrevention/CollisionPrevention.cpp

```diff
@@ -153,9 +263,11 @@ void CollisionPrevention::calculateConstrainedSetpoint(Vector2f &setpoint,
 			}
 		}
 
-	} else if (_last_message + MESSAGE_THROTTLE_US < hrt_absolute_time()) {
-		mavlink_log_critical(&_mavlink_log_pub, "No range data received");
-		_last_message = hrt_absolute_time();
+	} else {
+		// if distance data are stale, switch to Loiter and disable Collision Prevention
+		// such that it is still possible to fly in Position Control Mode
+		_publishVehicleCmdDoLoiter();
```

Comment:
Not sure if this is the best way to handle this.

Reply:
@mhkabir what do you suggest?

Reply:
IMO this is the best way. If your sensors fail you don't want the drone to continue flying as per normal. Ideally I think this should happen only once per arm, but if that will complicate the architecture a lot this is fine, it just means that you need to disable collision prevention in flight if the sensors fail.